pkg/dependencies/factory.go

```diff
@@ -953,6 +958,10 @@ func (f *Factory) WithPackageControllerClient(spec *cluster.Spec, kubeConfig str
 
 		httpProxy, httpsProxy, noProxy := getProxyConfiguration(spec)
 		eksaAccessKeyId, eksaSecretKey, eksaRegion := os.Getenv(config.EksaAccessKeyIdEnv), os.Getenv(config.EksaSecretAccessKeyEnv), os.Getenv(config.EksaRegionEnv)
+		endpoints := (map[string]string)(nil)
```

Comment:
```suggestion
		var endpoints map[string]string
```

Reply:
We're also passing this on to other things. This seems a little dangerous also. We're passing this map onward to other data structures. If those data structures try to append something the program will panic. It seems like this would be safer if it were initialized with `make(map[string]string)`.

Reply:
The map sets to `nil` in case of no registry mirror specified in the cluster spec, so making it as a empty map and leaving possibility of adding elements to it later is not logically intended.